torchao/quantization/quant_api.py

```diff
@@ -665,9 +666,15 @@ def apply_int4_weight_only_quant(weight):
         quant_min = 0
         quant_max = 15
         eps = 1e-6
-        preserve_zero = False
+        preserve_zero = zero_point_domain == ZeroPointDomain.INT
```

Comment:
this is probably not longer valid, since we use zero_point_domain == None by default so user don't need to worry about it. to make this clearer I'd suggest the following:

```
assert layout in layout_to_default_zero_point.keys(), f"Only support layout: {layout_to_default_zero_point.keys()}"
if zero_point_domain is None:
    zero_point_domain = layout_to_default_zero_point[layout]
    preserve_zero = layout_to_default_preserve_zero[layout]
else:
    assert zero_point_domain  in layout_to_supported_zero_point_domain[layout], "...."
```

Reply:
Could you check https://github.com/pytorch/ao/pull/1401/commits/17b8bf1ac39ea741c431afbb088109b2123781cd?